pkg/edgehub/config/config_test.go

```diff
@@ -0,0 +1,508 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"reflect"
+	"testing"
+	"time"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/kubeedge/kubeedge/pkg/common/util"
+)
+
+type testYamlGenerator struct {
+	Edgehub edgeHubConfigYaml `yaml:"edgehub"`
+}
+type testHeaderYamlGenerator struct {
+	SystemInfo extendHeaderConfigYaml `yaml:"systeminfo"`
+}
+
+type webSocketConfigYaml struct {
+	Url              string `yaml:"url,omitempty"`
+	CertFilePath     string `yaml:"certfile,omitempty"`
+	KeyFilePath      string `yaml:"keyfile,omitempty"`
+	HandshakeTimeout string `yaml:"handshake-timeout,omitempty"`
+	ReadDeadline     string `yaml:"read-deadline,omitempty"`
+	WriteDeadline    string `yaml:"write-deadline,omitempty"`
+}
+
+type extendHeaderConfigYaml struct {
+	Arch          string `yaml:"architecture,omitempty"`
+	DockerRootDir string `yaml:"docker_root_dir,omitempty"`
+}
+
+type controllerConfigYaml struct {
+	HeartbeatPeroid string `yaml:"heartbeat,omitempty"`
+	RefreshInterval string `yaml:"refresh-ak-sk-interval,omitempty"`
+	CloudhubURL     string `yaml:"cloud-hub-url"`
+	AuthInfosPath   string `yaml:"auth-info-files-path,omitempty"`
+	PlacementUrl    string `yaml:"placement-url,omitempty"`
+	ProjectID       string `yaml:"project-id,omitempty"`
+	NodeId          string `yaml:"node-id,omitempty"`
+}
+
+type edgeHubConfigYaml struct {
+	WSConfig  webSocketConfigYaml  `yaml:"websocket"`
+	CtrConfig controllerConfigYaml `yaml:"controller"`
+}
+
+//This function is used to create a temporary YAML file to be used for testing
+func generateTestYaml(test interface{}) {
+	data, err := yaml.Marshal(test)
+	if err != nil {
+		fmt.Print("Error in marshalling into temporary yaml data: ", err)
```

Comment:
Let this function return error instead of printing so that in test file we can verify whether this function was a success. Applicable to removeTemporaryDirectory

Reply:
Also put both these functions in pkg/common/util/config.go for reuse in other UT's